.ci/scripts/calculate_jobs.py

```diff
@@ -29,11 +29,12 @@ def set_output(key: str, value: str):
 
```

Comment:
https://github.com/matrix-org/synapse/actions/runs/5416759930/jobs/9848582611?pr=15851
```
  ERROR: Package 'matrix-synapse' requires a different Python: 3.7.13 not in '<4.0.0,>=3.8.0'
  
  Notice:  A new release of pip available: 22.2.2 -> 23.1.2
  Notice:  To update, run: pip install --upgrade pip
  Error: Command ['pip', 'install', '/tmp/cibuildwheel/repaired_wheel/matrix_synapse-1.87.0rc1-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl'] failed with code 1. 
```

Reply:
It looks like we're using the [x-range which uses the "latest stable version of Python"](https://github.com/actions/setup-python/blob/0d5da6a89a3aa5b4dbd48e38b0c9d678b0caa0e5/docs/advanced-usage.md#using-the-python-version-input)

https://github.com/matrix-org/synapse/blob/13fc89148c37f6c653dd237377ff165249fa73cc/.github/workflows/release-artifacts.yml#L33

https://github.com/matrix-org/synapse/blob/13fc89148c37f6c653dd237377ff165249fa73cc/.github/workflows/release-artifacts.yml#L79

https://github.com/matrix-org/synapse/blob/13fc89148c37f6c653dd237377ff165249fa73cc/.github/workflows/release-artifacts.yml#L128-L130

I guess, we would have to update this to `3.8` in order to move forward

Reply:
Actually, just needed to update `cp37` to `cp38` on this line:

https://github.com/matrix-org/synapse/blob/13fc89148c37f6c653dd237377ff165249fa73cc/.github/workflows/release-artifacts.yml#L147